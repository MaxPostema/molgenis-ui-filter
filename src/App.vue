<template>
  <div
    id="app"
    class="container"
  >
    <div class="row">
      <div class="col col-md-3">
        <FilterContainer
          v-model="selections"
          :filters="filters"
        />
      </div>
      <div class="col col-md-8">
        <pre>{{ selections }}</pre>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import * as components from './components'

export default Vue.extend({
  name: 'App',
  components,
  data () {
    return {
      selections: {
        search: 'value',
        country: ['nl']
      },
      filters: [ {
        name: 'search',
        label: 'Search',
        description: 'search by name',
        initiallyCollapsed: true,
        placeholder: 'test',
        type: 'string-filter'
      }, {
        name: 'smoking',
        label: 'Smoking',
        initiallyCollapsed: false,
        options: [{ value: true, text: 'Yes' }, { value: false, text: 'No' }, { value: null, text: 'N/A' }],
        type: 'checkbox-filter'
      }, {
        name: 'materials',
        label: 'Materials',
        initiallyCollapsed: false,
        options: [],
        maxVisibleOptions: 4,
        type: 'checkbox-filter'
      }, {
        name: 'country',
        label: 'Countries',
        initiallyCollapsed: false,
        options: [{ value: 'value', text: 'label' }, { value: 'nl', text: 'Nederland' }, { value: 'de', text: 'Duitseland' }],
        type: 'checkbox-filter'
      }, {
        name: 'biobank_quality',
        label: 'Biobank quality marks',
        initiallyCollapsed: false,
        options: [],
        maxVisibleOptions: 4,
        type: 'checkbox-filter'
      }, {
        name: 'collection_quality',
        label: 'Collection quality marks',
        initiallyCollapsed: false,
        options: [],
        maxVisibleOptions: 4,
        type: 'checkbox-filter'
      }, {
        name: 'type',
        label: 'Collection Types',
        initiallyCollapsed: false,
        options: [],
        maxVisibleOptions: 4,
        type: 'checkbox-filter'
      }, {
        name: 'dataType',
        label: 'Data types',
        initiallyCollapsed: false,
        options: [],
        maxVisibleOptions: 4,
        type: 'checkbox-filter'
      }]
    }
  }
})
</script>

<style>
</style>
