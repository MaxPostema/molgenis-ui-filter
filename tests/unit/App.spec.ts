import { shallowMount } from '@vue/test-utils'
import App from '@/App.vue'

describe('StringFilter.vue', () => {
  const wrapper = shallowMount(App)

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
