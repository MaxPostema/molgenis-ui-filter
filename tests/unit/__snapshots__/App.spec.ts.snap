// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringFilter.vue matches the snapshot 1`] = `
<div
  class="container"
  id="app"
>
  <div
    class="row"
  >
    <div
      class="col col-md-3"
    >
      <filtercontainer-stub
        filters="[object Object],[object Object]"
        value="[object Object]"
      />
    </div>
     
    <div
      class="col col-md-8"
    >
      <pre>
        {
  "string": "",
  "name": [
    "value2"
  ]
}
      </pre>
    </div>
  </div>
</div>
`;
